issue_25/get_seq.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Oct 14 11:25:08 2020
+
+@author: Bátora Dániel
+"""
+
+import os 
+from Bio import SeqIO
+from typing import Union
+
+
+
+def get_seq(filepath:str,organism:Union[str, int],temp_dir:str) -> str:
+    
+    """
+    Selects DNA sequence of the desired organims,
+    outputs it into a FASTA file 
+    
+    Args: 
+        filepath(str): Path to the original FASTA file 
+        organism(Union[str, int]): organism short name(str) or taxon id(int)
+        temp_dir(str): Path to the temporary directory
+        
+    Returns: 
+        output_path(str): Path to the output file saved in the temporary directory
+    
+    """
+    
+    
+    
+    if not os.path.isdir(temp_dir): 
+        os.mkdir(temp_dir)
+    output_path = os.path.join(temp_dir, organism + ".fasta")
+    reads = []
+    with open(filepath,  "r", encoding = "utf-8") as handle:
+        for record in SeqIO.parse(handle, "fasta"):
+            reads.append(record)
+    
+    hits = []
+    for record in reads:
+        if (isinstance(organism, str) and record.description.split("|")[3] == organism) or (isinstance(organism, int) and record.description.split("|")[4] == str(organism)):
+                hits.append(record)
+    SeqIO.write(hits, output_path, "fasta")
```

Comment:
Instead of checking whether the temporary directory is there, you can just assume it's there. If it's NOT there, the code will throw an OSError (you should add this under section `Raises:` in the docstring). We want to separate concerns in different functions. If we check everywhere everything every function will get bloated and we will have a lot of repeat code. If the person using the function does not use it correctly an error is thrown, that's fair enough. So no need to try to correct their errors, we can take care of that upstream when the function is called.